src/App.jsx

```diff
@@ -14,33 +16,88 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+const SORT_BY_ALPHABET = 'alphabet';
+const SORT_BY_LENGTH = 'length';
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+export const App = () => {
+  const [goods, setGoods] = useState([...goodsFromServer]);
+  const [sortOrder, setSortOrder] = useState('');
+  const [isReversed, setIsReversed] = useState(false);
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+  const applySort = (sortType) => {
+    let sortedGoods = [...goodsFromServer];
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+    switch (sortType) {
+      case SORT_BY_ALPHABET:
+        sortedGoods.sort((a, b) => a.localeCompare(b));
+        break;
+      case SORT_BY_LENGTH:
+        sortedGoods.sort((a, b) => a.length - b.length);
+        break;
+      default:
+        sortedGoods = [...goodsFromServer];
+    }
+
+    if (isReversed) {
+      sortedGoods.reverse();
```

Comment:
The `isReversed` state is used to determine if the goods should be reversed after sorting. However, the `applySort` function sets the `sortOrder` state without considering the current `isReversed` state. This means that if the list is reversed and then a new sort is applied, the list will be sorted but not reversed again. Consider updating the `isReversed` state or handling the reversal logic more explicitly when applying a new sort.